mainwindow.cpp

```diff
@@ -56,35 +59,15 @@ void MainWindow::on_goButton_clicked()
         ui->player2Label->setText("SCISSORS");
         break;
     }
-    ui->resultLabel->setText(get_result(player1Choice, player2Choice));
-    qDebug() << "It's " << player1Choice << " vs " << player2Choice;
+    ui->resultLabel->setText(get_result(player1->choice, player2->choice));
+    qDebug() << "It's " << player1->choice << " vs " << player2->choice;
     ui->paperButton->setEnabled(true);
     ui->rockButton->setEnabled(true);
     ui->scissorsButton->setEnabled(true);
     display_result();
     roundCount++;
 }
 
-void MainWindow::display_choice()
-{
-    switch (player1Choice) {
-    case ROCK:
-        ui->player1Label->setText("ROCK");
-        break;
-    case PAPER:
-        ui->player1Label->setText("PAPER");
-        break;
-    case SCISSORS:
-        ui->player1Label->setText("SCISSORS");
-        break;
-    }
-}
-
-Choices MainWindow::generate_choice()
-{
-    return static_cast<Choices>(rand() % 3);
-}
-
 QString MainWindow::get_result(Choices p1, Choices p2)
 {
     if (p1 == p2)
```

Comment:
why we need to calculate winCount?

Reply:
what do you mean?
I want to have winCount, loseCount and tieCount, but roundCount =  winCount + loseCount + tieCount
so instead of increasing winCount, I just calculate it through tieCount  and loseCount

Reply:
I mean it is difficult to track the code flow when you doing like this, my suggest is to move this logic to where you handle tieCount and loseCount

Reply:
hmm ok, imma move it somewhere else